tests/ui/no_generics.rs

```diff
@@ -0,0 +1,9 @@
+#[derive_where::derive_where(Clone)]
+struct Test1(u8);
+
+// Make sure this doesn't reach unreachable code as it should fail early for
+// not having any generics.
```

Comment:
I don't disagree with erroring on these, as the standard derive is always preferable if possible, but I don't think there is unreachable code to reach

Reply:
This test is making sure that this line is never reached:
https://github.com/ModProg/derive-where/blob/d41f608e029f136e734800767d8a978ea1735c6e/src/lib.rs#L184
It should never be reached because a check happens first to check for any generics. Rust is smart enough to find no generics if they aren't used inside the item.

Maybe the comment needs to be improved?

Reply:
oh ok, no it's ok. 
Even without the unreachable, we just don't want to support it anyways because it doesn't make sense. We can keep it like that.